test-integ/peering_commontopo/README.md

```diff
@@ -1,12 +1,66 @@
-# peering_commontopo
+# CONSUL PEERING COMMON TOPOLOGY TESTS
 
 These peering tests all use a `commonTopo` (read: "common topology") to enable sharing a deployment of a Consul. Sharing a deployment of Consul cuts down on setup time.
 
-This is only possible if two constraints are followed:
+To run these tests, you will need to have docker installed. Next, make sure that you have all the required consul containers built:
 
+```
+make test-compat-integ-setup
+```
+
+## Non-Shared CommonTopo Tests
+
+The tests in question are designed in a manner that modifies the topology. As a result, it is noy possible to share the testing environment across these tests.
+
+## Shared CommonTopo Tests
+
+The tests in question are designed in a manner that does not modify the topology. As a result, it is possible to share the testing environment across these tests.
+
+To run all consul peering tests with no shared topology, run the following command:
+
+```
+cd /path/to/peering_commontopo
+go test -timeout=10m -v -no-share-topo . 
+```
+
+To run all peering tests with shared topology only:
+
+```
+cd /path/to/peering_commontopo
+go test -timeout=10m -run ^'TestSuitesOnSharedTopo' -v . 
+```
+
+To run individual peering topology tests:
+
+```
+cd /path/to/peering_commontopo
+go test -timeout=10m -run ^'ExampleTestSuite' -v .    
+```
+
+## Local Development and Testing
```

Comment:
Hm, this whole section could use a rewrite. For non-shareable suites, we don't need to implement `sharedTopoSuite`. They don't even really need to be structured with a discrete `setup` and `test` method strictly speaking. They